api.go

```diff
@@ -716,11 +718,12 @@ func (r *Raft) BootstrapCluster(configuration Configuration) Future {
 // Leader is used to return the current leader of the cluster.
 // It may return empty string if there is no current leader
 // or the leader is unknown.
-func (r *Raft) Leader() ServerAddress {
+func (r *Raft) Leader() (ServerAddress, ServerID) {
```

Comment:
I believe this would be considered a breaking change of the library interface. I imagine we don't want to release a v2 for this, so I suggest that we keep the old method, add a `Deprecated: ...` line as the last line of the godoc, and add a new `LeaderWithID()` method which returns the extra data.

The deprecated `Leader` method can call `LeaderWithID` and return only the one value.

Reply:
Yes we will need to do this or a v2. 

@mkeeler and I had a conversation about it and we where aligning for a v2 but looking at this a bit more I'm not sure it's worth doing a v2 only for this change?
What you think @mkeeler?